migrations/20170806154824_user_prefs.js

```diff
@@ -0,0 +1,20 @@
+export async function up(knex) {
+  await knex.schema.table('users', (table) => {
+    table.jsonb('preferences').defaultTo('{}').notNullable();
+  });
+  // Migrate from the 'frontend_preferences'
+  await knex.raw(`
+    update users set preferences = jsonb_set(
+      preferences,
+      '{hideCommentsOfTypes}',
+      coalesce((frontend_preferences::jsonb) #> '{net.freefeed,comments,hiddenTypes}', '[]'),
+      true
+    )
+  `);
+}
+
+export async function down(knex) {
+  await knex.schema.table('users', (table) => {
+    table.dropColumn('preferences');
+  });
```

Comment:
Strictly speaking, we should copy values from `preferences` to `frontend_preferences` before dropping the column